src/core/prompts/sections/custom-instructions.ts

```diff
@@ -30,6 +48,17 @@ export async function loadRuleFiles(cwd: string): Promise<string> {
 	return combinedRules
 }
 
+async function findCustomInstructionsFile(dir: string, filePattern: string): Promise<string> {
+	// First try to find as a direct file
+	const content = await findRuleInDirectory(dir, filePattern)
```

Comment:
Unfortunately I just realized that this crashed my extension with a `Error: ENAMETOOLONG: name too long, open` when the "filePattern" is actually a long string of instructions. Maybe we need to more explicitly differentiate between filenames and content here? I'm going to need to revert this for now though as it's making the extension not work for me. Let me know if you want to discuss how to bring it back safely!

Reply:
@mrubens ,  can you share the failure case with more details? I want to repro and fix it

Reply:
Yes, it seems to happen if I send any message as Architect with the built-in prompt. Looks like because it's interpreting the long custom instructions as a potential filename and getting a `ENAMETOOLONG` error. Does that help? Let me know if not.

![Screenshot 2025-03-25 at 9 33 13 AM](https://github.com/user-attachments/assets/65ed514e-3c47-46e0-8631-431d354780c1)